### DISCUSSION TITLE
Both installation methods generate errors

### DISCUSSION THREAD

**Discussion by stephennutt** [zachowj/node-red-contrib-home-assistant-websocket#1280] [Q&A]
I have been using this pallet for a couple of years and don't remember un-installing it.  After noticing it was gone, I went to reinstall using GUI and got this error message in logs:

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/25011547/2d0b7236-7a02-459e-83e4-d4d4abe8527f)

When I try to install using NPM, I get these messages:

![image](https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/25011547/db4bbc9f-7214-4e7d-be71-c3fabe243b46)

Any suggestions on how to fix for a Linux newbie?

**Comment by zachowj** [comment]
Are you running at least version 3.1.1 of Node-RED?

**Comment by stephennutt** [comment]
I was running 3.1.0 so I went to upgrade using script.  However, while upgrading, I got the error  "Bad Install for Node.js".  I did a apt update and upgrade, then tried the script again.  That time it installed with no errors.  When I opened Node Red, node-red-contrib-home-assistant-websocket was already installed. Weird!  It seems to be working now.  Thanks for the simple question.
